### PR TITLE
Drop support for python 3.6

### PR DESCRIPTION
**Pull request by killjoy1221**
Python 3.6 is end of life.

Also added 3.10 to tox and gh-actions

### CODE REVIEW

**Comment by NickCrews** [comment]
Hey @killjoy1221, thanks for this! This is totally the right direction. A few minor tweaks:

- Please remove the commit that adds to `.gitignore`. If you want to ignore those files, do it in your [personal global .gitignore](https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/). This .gitignore should stay as lean and specific to this repo as possible
- Looks like CI fails because python 3.10 is getting interpreted as a float. You need to quote it so it's a string
- Thanks for updating the CHANGELOG, that's something that's easy to not do! Can you add that we start testing for python 3.10 there as well?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.7, 3.8, 3.9, 3.10]
```
Comment (NickCrews) [inline .github/workflows/build.yml:21]: Should be `"3.10"`, not just `3.10` which is interpreted as the float `3.1`

File: CHANGELOG.rst [hunk 2]
```diff
@@ -4,6 +4,12 @@ Changelog
 `Unreleased <https://github.com/staticjinja/staticjinja/compare/main>`_
 -----------------------------------------------------------------------
 
+Changed
+^^^^^^^
+
+* Drop support for Python 3.6. See
```
Comment (NickCrews) [inline CHANGELOG.rst:10]: Can you actually phrase this as "Stop testing for Python 3.6" It still might work, it's just not officially supported.

**Comment by killjoy1221** [comment]
Thanks for the feedback. I think I got everything fixed up.

I've also got a branch adding typing everywhere. Would you be interested in a PR for that?

**Comment by killjoy1221** [comment]
I think black failed because there's a version conflict with click. As a solution, I can either undo the `poetry update` and leave that for a separate commit/PR or update black to 22.3.0. `poetry.lock` specifies `black-21.12b0`

**Comment by NickCrews** [comment]
That's my read on why black failed as well. Upgrading black to `black = { version = "^22.3.0", optional = true }` and re-running `poetry update` would be awesome if you're willing.

**Comment by NickCrews** [comment]
Thanks @killjoy1221! Do you need a release?

A PR adding typing would be great!

**Pull request merged**